File: Daily Challenge/January 2025/Maximum Score After Splitting a String.cpp
class Solution {
public:
  int maxScore(string s) {
    int countOne = 0, countZero = 0;
    for (char c : s) {
      if (c == '1') countOne++;
    }

    int maxScore = 0;
    for (int i = 0; i < s.size() - 1; i++) {
      if (s[i] == '0') countZero++;
      else countOne--;
      maxScore = max(maxScore, countZero + countOne);
    }

    return maxScore;
  }
};

File: Daily Challenge/January 2025/Maximum Score After Splitting a String.cs
public class Solution {
  public int MaxScore(string s) {
    var countOne = 0;
    var countZero = 0;
    foreach (var c in s) {
      if (c == '1') countOne++;
    }

    var maxScore = 0;
    for (var i = 0; i < s.Length - 1; i++) {
      if (s[i] == '0') countZero++;
      else countOne--;
      maxScore = Math.Max(maxScore, countZero + countOne);
    }

    return maxScore;
  }
}